File: minimum-add-to-make-parentheses-valid/minimum-add-to-make-parentheses-valid.java
class Solution {
public int minAddToMakeValid(String s) {
    int countleft=0;
    int count=0;
    for(int i=0;i<s.length();i++){
        if(s.charAt(i)=='(')
            ++countleft;
        else{
            if(countleft<=0)
                count++;
           else{
            --countleft;
           }}}
        return count+countleft;
    }
}

File: 957-minimum-add-to-make-parentheses-valid/minimum-add-to-make-parentheses-valid.py
class Solution:
    def minAddToMakeValid(self, s: str) -> int:
        closed = 0
        opened = 0
        for i in s:
                if i == '(':
                    opened += 1
                    print(i)
                    print(opened)
                elif i == ')' and opened > 0:
                    opened -= 1
                    print(i)
                    print(opened)
                elif i == ')' and opened <= 0:
                    closed += 1
                    print(i)
                    print(opened)
        return opened + closed